referenceframe/frame.go

```diff
@@ -198,6 +198,22 @@ func (nf *noGeometryFrame) Geometries(input []Input) (*GeometriesInFrame, error)
 	return NewGeometriesInFrame(nf.Name(), nil), nil
 }
 
+// namedFrame is used to change the name of a frame.
+type namedFrame struct {
```

Comment:
[q] why do we we want this to remain unexported? If a user is implementing their own modular arm would they need access to this struct?

Reply:
Most frame constructors allow the caller to provide a name for the returned frame. There are only a few cases where a pre-made frame would need a different name than the one it already has.

Reply:
> Is there a risk to not being able to rename - is there the possibility of this making things harder to debug or even hurt a user experience?

Very much the opposite. If a frame can change names whenever it likes, tracking it while degbugging becomes much more difficult.

Calling NewNamedFrame is less of a "workaround" and more the intended workflow. There's really no need to directly instantiate the struct rather than using the constructor.